runtime/lua/vim/diagnostic.lua

```diff
@@ -755,7 +755,31 @@ function M.get(bufnr, opts)
     opts = { opts, 't', true },
   })
 
-  return get_diagnostics(bufnr, opts, false)
+  return vim.deepcopy(get_diagnostics(bufnr, opts, false))
```

Comment:
Moving the `deepcopy` out of `get_diagnostics` and into `vim.diagnostic.get()` is a good change regardless, and should have been in the first place. There are a lot of call sites of `get_diagnostics` where a copy is not needed at all, so we've been wastefully copying diagnostics this whole time.

Reply:
Yeah, I also realized this during making the PR.